ParticleObjects/Source/pointSource_class.f90

```diff
@@ -210,9 +226,16 @@ subroutine sampleEnergy(self, p, rand)
     class(pointSource), intent(inout)   :: self
     class(particleState), intent(inout) :: p
     class(RNG), intent(inout)           :: rand
+    real(defReal)                       :: r
+    integer(shortInt)                   :: g
 
     if (self % isMG) then
-      p % G = self % G
+      r = rand % get()
```

Comment:
```suggestion
      !! TODO: Replace this single, common procedure to sample discrete distributions
      r = rand % get()
```
Sadly the Alias method is implemented... but in the infamous photon branch... :-(